apps/event-worker/src/workers/update-resource-variable.ts

```diff
@@ -29,7 +30,12 @@ export const updateResourceVariableWorker = createWorker(
           eq(schema.deploymentVariable.key, key),
         ),
       )
-      .then((results) => results.map((result) => result.deployment.id));
+      .then((results) => ({
+        deploymentIds: results.map((result) => result.deployment.id),
+        resource: results[0]?.resource,
+      }));
```

Comment:
no, make this different, const resource = results.at(0) 
You are making this way more complicated